blueflood-core/src/test/java/com/rackspacecloud/blueflood/types/BluefloodTimerRollupPercentileTest.java

```diff
@@ -0,0 +1,220 @@
+package com.rackspacecloud.blueflood.types;
+
+import org.junit.Test;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+import static org.junit.Assert.*;
+
+public class BluefloodTimerRollupPercentileTest {
+
+    @Test
+    public void maybePromoteDoubleYieldsSameValue() {
+        // given
+        Double value = 123.45d;
+        // when
+        Number result = BluefloodTimerRollup.Percentile.maybePromote(value);
+        // then
+        assertNotNull(result);
+        assertEquals(Double.class, result.getClass());
+        assertEquals(123.45d, (Double)result, 0.00001d);
+    }
+
+    @Test
+    public void maybePromoteFloatPromotesToDouble() {
+        // given
+        Float value = 123.45f;
+        // when
+        Number result = BluefloodTimerRollup.Percentile.maybePromote(value);
+        // then
+        assertNotNull(result);
+        assertEquals(Double.class, result.getClass());
+        assertEquals(123.45d, (Double)result, 0.00001d);
+    }
+
+    @Test
+    public void maybePromoteLongYieldsSameValue() {
+        // given
+        Long value = 123L;
+        // when
+        Number result = BluefloodTimerRollup.Percentile.maybePromote(value);
+        // then
+        assertNotNull(result);
+        assertEquals(Long.class, result.getClass());
+        assertEquals(123L, (long)(Long)result);
```

Comment:
@shintasmith According to [this](http://stackoverflow.com/questions/36869186/what-is-the-difference-between-casting-to-a-long-and-callint-longvalue), they are equivalent.